### PR TITLE
LLM: Add one-click installer for Windows

### PR DESCRIPTION
**Pull request by JinBridger**
## Description

Add one-click installer for Windows.

- Tested on ChatGLM2-6b, Baichuan-13B-Chat, Baichuan2-7B-Chat
- Tested on another Windows machine without Python.

For usage, please refer to README.md.

### CODE REVIEW

File: python/llm/portable-installer/chat.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+def stream_chat(model,
+                tokenizer,
+                stopping_criteria,
+                input_str,
+                chat_history):
+    prompt = format_prompt(input_str, chat_history)
+    # print(prompt)
+    input_ids = tokenizer([prompt], return_tensors="pt")
+    streamer = TextIteratorStreamer(tokenizer, skip_prompt=True, skip_special_tokens=True)
+    generate_kwargs = dict(input_ids, streamer=streamer, max_new_tokens=512, stopping_criteria=stopping_criteria)
```
Comment (Oscilloscope98) [inline python/llm/portable-installer/chat.py:54]: `use_cache=True` should be added into generation config, not in the `from_pretrained` function. Refer to [here](https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationConfig.use_cache)

File: python/llm/portable-installer/chat.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                                                    low_cpu_mem_usage=True,
+                                                    torch_dtype="auto",
+                                                    trust_remote_code=True,
+                                                    use_cache=True)
+    else:
+        return AutoModel.from_pretrained(model_path,
+                                         low_cpu_mem_usage=True,
+                                         torch_dtype="auto",
+                                         trust_remote_code=True,
+                                         use_cache=True)
```
Comment (Oscilloscope98) [inline python/llm/portable-installer/chat.py:84]: Maybe we should add a try except logic here to catch the case for the model which cannot be loaded by `AutoModelForCausalLM` and `AutoModel`, and print a related message.

Later we could find a better way to support more model class

File: python/llm/portable-installer/setup.bat [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+:: enable run site.main() automatically
+cd .\python-embed
+set "search=#import site"
+set "replace=import site"
+powershell -Command "(gc python39._pth) -replace '%search%', '%replace%' | Out-File -encoding ASCII python39._pth"
+cd ..
+
+:: install pip packages
+%python-embed% -m pip install bigdl-llm[all] transformers_stream_generator
```
Comment (Oscilloscope98) [inline python/llm/portable-installer/setup.bat:20]: Maybe also add dependency: `tiktoken` and `einops`.

`einops` is required for mpt, falcon and qwen;
`tiktoken` is required for qwen

File: python/llm/portable-installer/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# BigDL-LLM One-Click Installer For Windows
+
+## Supported models
+
+- ChatGLM2-6b
+- Baichuan-13B-Chat
+- Baichuan2-7B-Chat
```
Comment (Oscilloscope98) [inline python/llm/portable-installer/README.md:7]: We could change the title to `Verified Models`, as I believe this one-click tool can work for much more models.

File: python/llm/portable-installer/setup.bat (hunk 3 above)
Comment (shane-huang) [reply python/llm/portable-installer/setup.bat:20]: I think we'd better separate the setup scripts/README and user scripts&README. We use setup.bat to build the zip, and the zip should only contain the user scripts and user guide.

Comment (shane-huang) [reply python/llm/portable-installer/setup.bat:20]: In user guide please add a step-by-step usage guide with screen snapshots.

Comment (shane-huang) [reply python/llm/portable-installer/setup.bat:20]: Also, you can upload a gif of the process in github to show the entire process.

File: python/llm/portable-installer/chat.py (hunk 2 above)
Comment (JinBridger) [reply python/llm/portable-installer/chat.py]: Added.

File: python/llm/portable-installer/README.md (hunk 4 above)
Comment (JinBridger) [reply python/llm/portable-installer/README.md:7]: Done.

**Comment by Oscilloscope98** [comment]
It would be better to change the question in the demo gif :)

For example, change "谁创造了你" to "AI是什么“ etc. I feel a little strange to see the name of another company in the demo video.

**Comment by JinBridger** [comment]
> It would be better to change the question in the demo gif :)
> 
> For example, change "谁创造了你" to "AI是什么“ etc. I feel a little strange to see the name of another company in the demo video.

Yes, I recorded another gif and now there's no company name in the gif.

File: python/llm/portable-installer/setup.bat (hunk 3 above)
Comment (JinBridger) [reply python/llm/portable-installer/setup.bat]: Already done as you said  :)

**Comment by jason-dai** [comment]
Do not upload pictures to this repo; see gif examples at https://github.com/intel-analytics/BigDL/blob/main/README.md

File: python/llm/portable-installer/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# BigDL-LLM One-Click Installer For Windows: User Guide
```
Comment (shane-huang) [inline python/llm/portable-installer/README.md:1]: Since we actually didn't require user to install anything, shall we call it "BigDL-LLM Portable"?

Comment (JinBridger) [reply python/llm/portable-installer/README.md]: Already done as you said. :)

File: python/llm/portable-executable/setup.bat [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+:: enable run site.main() automatically
+cd .\python-embed
+set "search=#import site"
+set "replace=import site"
+powershell -Command "(gc python39._pth) -replace '%search%', '%replace%' | Out-File -encoding ASCII python39._pth"
+cd ..
+
+:: install pip packages
+%python-embed% -m pip install bigdl-llm[all] transformers_stream_generator tiktoken einops
```
Comment (Oscilloscope98) [inline python/llm/portable-executable/setup.bat:20]: Do we also need `colorama` here?

Comment (JinBridger) [reply python/llm/portable-executable/setup.bat]: It is installed as a dependency when install other packages. Anyway, it's better to add it. I've added it in the latest commit.

**Review by shane-huang — APPROVED** [review]
LGTM. We'll add GPU version in another PR.

**Comment by jason-dai** [comment]
Is there an installation/setup guide? @shane-huang @Oscilloscope98

**Comment by Oscilloscope98** [comment]
> Is there an installation/setup guide? @shane-huang @Oscilloscope98

We currently have a very simple setup-guide for how to build the zip file: https://github.com/intel-analytics/BigDL/blob/main/python/llm/portable-executable/setup.md

**Pull request merged**